### ISSUE TITLE
Pidgin default status support gone?

### ISSUE THREAD

**Issue by GoogleCodeExporter** [dequis/msn-pecan#104]
```
What steps will reproduce the problem?
1. Disable the personal message feature of Pecan
2. Use the normal status area in Pidgin to create the personal message
3. Look at your message not apply to your name.

What is the expected output? What do you see instead?
The personal message should work like before if you disabled Pecan's new
ability.

What version of the product are you using? On what operating system?
0.0.17 on Ubuntu 8.10

Apparently Pidgin's slightly more limited way of personal message/status
combo works better for me, since I use the Autoreply plugin which sends the
custom message according to that status. But disabling the Pecan support,
and having myself added to my buddy list, I don't see the status applied to
my personal message anymore.


```

Original issue reported on code.google.com by `mastaste...@gmail.com` on 6 Dec 2008 at 8:26

**Comment by GoogleCodeExporter** [comment]
```
As I can see disabling "independent pm" you cannot set the psm. I know where is 
the
problem and I'm working on a fix for this, maybe it will not good but will work.
```

Original comment by `devid...@gmail.com` on 8 Dec 2008 at 9:25
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
I've just committed on http://github.com/dfiloni/msn-pecan a patch to fix this
(http://github.com/dfiloni/msn-pecan/commit/c4de2485ba69f0079c355ab9aa3104b4f78c
5053), can
you try it?

Felipe, can you also check if the patch is ok or not? I honestly don't like it, 
but
seems to work fine for me.
```

Original comment by `devid...@gmail.com` on 10 Dec 2008 at 6:41

**Comment by GoogleCodeExporter** [comment]
```
Ah, very interesting patch! Good idea.

But why not do it the other way around? You add first the statuses without the
message, and then if "use_independent_pm" is not set, then you add the message 
with
purple_status_type_add_attrs.
```

Original comment by `felipe.contreras` on 10 Dec 2008 at 7:14
- Added labels: **Type-Defect**, **Priority-Critical**

**Comment by GoogleCodeExporter** [comment]
```
Already tried, it was my first idea, it doesn't work well. It shows an error 
message:
attempt to set status attribute "message" for status which is not legal...

The committed fix works fine as I can see.
```

Original comment by `devid...@gmail.com` on 10 Dec 2008 at 7:36

**Comment by GoogleCodeExporter** [comment]
```
Are you sure? It looks like that message an only be triggered by calling some
purple_status_set_active* function, not purple_status_type_add_attrs.
```

Original comment by `felipe.contreras` on 10 Dec 2008 at 7:48

**Comment by GoogleCodeExporter** [comment]
```
I've also tried with the purple_status_set_active if I'm not wrong...
```

Original comment by `devid...@gmail.com` on 10 Dec 2008 at 7:51

**Comment by GoogleCodeExporter** [comment]
```
This should be fixed in 0.0.18.
```

Original comment by `felipe.contreras` on 15 Feb 2009 at 11:37
- Changed state: **Fixed**
